Add documentation for hydra.job.override_dirname (#1448)
facebookresearch/hydra: website/docs/configure_hydra/job.md

You can override it via the command line, or your config file. 

### hydra.job.override_dirname

> This config enables creating working dir based on command line overrides. 
> For more information, check [here](/configure_hydra/workdir.md).

Reviewer: Phrasing is a but clunky.
```suggestion
Enables the creation of an output directory which is based on command line overrides.
See [this](/configure_hydra/workdir.md) for more infromation.
```

Author: fixed

```
With this configuration, running
```bash

> $ python my_app.py --multirun a=10 b=20 seed=777,999

Reviewer: sweep over `a=a1,a2 b=b1 seed=1,2` to show more useful output.

```

> Would result in a directory like:

Reviewer: ```suggestion
Would result in a directory structure like:
```

Author: will do!

```

> $ tree multirun/

Reviewer: use `tree multirun -d` to list directories only.
